### PR TITLE
Create explain.js

### PR DESCRIPTION
**Pull request by craigspaeth**

### CODE REVIEW

File: explain.js [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+      </select>
+    );
+  }
+}
+
+class ArtworkVisibility extends React.Component {
+  // ...
+  onToggleVisibility() {
+    $.ajax({
+      url: `/artwork/${this.props.id}/request_to_publish`,
```
Comment (craigspaeth) [inline explain.js]: We would try to push as much "model" logic to Rails as possible. For instance in this case we have some functionality that involves "requesting to publish" an artwork—we might decide to cater an endpoint in Rails that just accepts the artwork `id` instead of PUTing generically a blob of artwork data.

Likewise [here](https://github.com/craigspaeth/react-rails-explain/pull/2/files#diff-a2341bfa113d54962ff47644b614e4d7R5) we might accept data such as `price_visibility: 'forsale'` which doesn't 1:1 reflect the data model of Gravity but it directly maps to the UI's use case.

Basically trying to say any logic that can be moved to endpoints server-side should be—and React should be responsible for a little view-only logic as it can be.

File: explain.js [hunk 2]
```diff
@@ -0,0 +1,53 @@
+class ArtworkAttrSelect extends React.Component {
+  onChange(e) {
+    this.setState({ disabled: true })
+    $.ajax({
```
Comment (craigspaeth) [inline explain.js]: The form helpers are nice from Rails, but I wonder how much value we're really getting from them when a simple `$.ajax` would work.

Rails' convention-over-config/helper-laden ways tend to work well because Rails has full-stack control from database to routing to filesystem to view. When moving to the client-side you lose a lot of that and have to start introducing [more](https://github.com/railsware/js-routes) and [more](https://github.com/artsy/volt/pull/1792/files#diff-59896739fefb0ad9e13ed224720991a8R5) [glue](https://github.com/artsy/volt/pull/1792/files#r62237306) abstractions to combine the magic of Rails into JS world.

File: explain.js [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+    id={$el.attr('data-id')}
+    attr='price_visibility'
+    options={['forsale', 'range', 'notforsale']}
+  />
+  const visibilityLink = <ArtworkVisibility
+    id={$el.attr('data-id')}
+  />
+  React.render(availabilitySelect, $el.find('.availability-container'))
+  React.render(priceSelect, $el.find('.price-container'))
+  React.render(visibilityLink, $el.find('.visibility-container'))
```
Comment (craigspaeth) [inline explain.js]: In most cases we could probably get away with cleanly separating the Rails views from the React parts. e.g. here we recognize that only these three empty elements rendered from Rails are where React components live. Not even going to bother piecing them together into a bigger React app—just render, essentially, three tiny decoupled React apps inside each list item.

A step after this might then involve combining these three components into a React list item component that takes over all of list rendering from Rails.

File: explain.js [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+      {_.map(this.props.options, (val) =>
+        <option value="{val}">{t(val)}</option>
+      )}
+      </select>
+    );
+  }
+}
+
+class ArtworkVisibility extends React.Component {
+  // ...
```
Comment (craigspaeth) [inline explain.js]: In this "keep React and Rails totally separate" one would write modal functionality in React and not try to wrap up remodal.js. The path forward might involve refactoring all modals into our own React modal component with the hope to drop all other libraries.
